core/module.h

```diff
@@ -318,6 +384,23 @@ class Module {
 
   virtual CheckConstraintResult CheckModuleConstraints() const;
 
+  /*!
+   * Whether the module overrides RunTask.
+   */
+  virtual bool IsTask() const { return false; };
```

Comment:
I just realized another way of doing this without a virtual function is to store "is task" as a bool member in Module and then just have a non-virtual accessor and have it be false by default.  Then task-implementing modules can just set it to true in their constructors.